src/Modal.js

```diff
@@ -267,6 +268,15 @@ const Modal = React.createClass({
       );
     }
 
+    const keys = Object.keys(_props);
+    const propTypes = Object.keys(Modal.propTypes);
+    const newProps = {};
+    keys.map((prop) => {
+      if (propTypes.indexOf(prop) === -1) {
+        newProps[prop] = _props[prop];
+      }
+    });
```

Comment:
can you hoist this out of hte render method as an `omitProps` function just so its a bit clearer what's going on?